### PR TITLE
Perf CI

### PR DESCRIPTION
**Pull request by ebattat**
Adding perf ci support:
* Create a dedicate folder for perf_ci
* Adding grafana dashboards

### CODE REVIEW

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]

File: .github/workflows/Nightly_Func_CI.yml [hunk 1]
```diff
@@ -32,21 +32,35 @@ jobs:
...
         ELASTICSEARCH_PORT: ${{ secrets.ELASTICSEARCH_PORT_CI }}
+        RUNNER_PATH: ${{ secrets.RUNNER_PATH }}
       run: |
         #Functional CI must update hosts ci file
         sudo echo "${{ secrets.HOSTS1_CI }}" | sudo tee -a /etc/hosts
         sudo echo "${{ secrets.HOSTS2_CI }}" | sudo tee -a /etc/hosts
+        # get last commit id of benchmark-operator
+        git clone https://github.com/cloud-bulldozer/benchmark-operator $RUNNER_PATH/benchmark-operator
+        cd $RUNNER_PATH/benchmark-operator
+        benchmark_operator=$(git log --format="%H" -n 1)
```
[the remark this answers is not in the document]
Comment (ebattat) [reply .github/workflows/Nightly_Func_CI.yml:43]: Thx for the input, update it in last commit

File: .github/workflows/Nightly_Func_CI.yml [hunk 2]
```diff
@@ -32,21 +32,35 @@ jobs:
...
+        cd -
+        # get last commit id of benchmark-wrapper
+        git clone https://github.com/cloud-bulldozer/benchmark-wrapper $RUNNER_PATH/benchmark-wrapper
+        cd $RUNNER_PATH/benchmark-wrapper
+        benchmark_wrapper=$(git log --format="%H" -n 1)
```
[the remark this answers is not in the document]
Comment (ebattat) [reply .github/workflows/Nightly_Func_CI.yml:48]: Thx for the input, update it in last commit

File: .github/workflows/Nightly_Func_CI.yml [hunk 3]
```diff
@@ -32,21 +32,35 @@ jobs:
...
+        cd -
         BUILD=$(pip freeze | grep benchmark-runner | sed 's/==/=/g')
         BUILD_VERSION="$(cut -d'=' -f2 <<<"$BUILD")"
+        start=`date +%s
```
[the remark this answers is not in the document]
Comment (ebattat) [reply .github/workflows/Nightly_Func_CI.yml:52]: Thx for the input, update it in last commit. add missing `

File: .github/workflows/Nightly_Func_CI.yml [hunk 4]
```diff
@@ -55,12 +69,25 @@ jobs:
...
+        benchmark_operator=$(git log --format="%H" -n 1)
+        cd -
+        # get last commit id of benchmark-wrapper
+        git clone https://github.com/cloud-bulldozer/benchmark-wrapper $RUNNER_PATH/benchmark-wrapper
+        cd $RUNNER_PATH/benchmark-wrapper
+        benchmark_wrapper=$(git log --format="%H" -n 1)
+        cd -
         BUILD=$(pip freeze | grep benchmark-runner | sed 's/==/=/g')
         BUILD_VERSION="$(cut -d'=' -f2 <<<"$BUILD")"
+        end=`date +%s`
```
[the remark this answers is not in the document]
Comment (ebattat) [reply .github/workflows/Nightly_Func_CI.yml:90]: Thx for the input, update it in last commit.

File: .github/workflows/Nightly_Perf_CI.yml [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        PIN_NODE1: ${{ secrets.PIN_NODE1 }}
+        PIN_NODE2: ${{ secrets.PIN_NODE2 }}
+        ELASTICSEARCH: ${{ secrets.ELASTICSEARCH }}
+        ELASTICSEARCH_PORT: ${{ secrets.ELASTICSEARCH_PORT }}
+        RUNNER_PATH: ${{ secrets.RUNNER_PATH }}
+      run: |
+        # get last commit id of benchmark-operator
+        git clone https://github.com/cloud-bulldozer/benchmark-operator $RUNNER_PATH/benchmark-operator
+        cd $RUNNER_PATH/benchmark-operator
+        benchmark_operator=$(git log --format="%H" -n 1)
```
[the remark this answers is not in the document]
Comment (ebattat) [reply .github/workflows/Nightly_Perf_CI.yml:42]: Thx for the input, update it in last commit.

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads.py [hunk 6]
```diff
@@ -19,39 +19,32 @@ class BenchmarkOperatorWorkloads:
     This class contains all the custom_workloads
     """
     def __init__(self, kubeadmin_password: str = '', es_host: str = '', es_port: str = ''):
-        self._runner_version = None
+        # environment variables
+        self.__environment_variables_dict = environment_variables.environment_variables_dict
```
[the remark this answers is not in the document]
Comment (ebattat) [reply benchmark_runner/benchmark_operator/benchmark_operator_workloads.py:23]: This dictionary holds all environment variable.

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads.py [hunk 7]
```diff
@@ -262,16 +255,18 @@ def get_metadata(self, kind: str = None, database: str = None):
             metadata.update({'db_version': 10.3})
         return metadata
 
-    def update_ci_status(self, status: str):
+    def update_ci_status(self, status: str, ci_minutes_time: int, benchmark_operator: str, benchmark_wrapper: str):
         """
         This method update ci status Pass/Failed
+        :param benchmark_wrapper:
+        :param benchmark_operator:
```
[the remark this answers is not in the document]
Comment (ebattat) [reply benchmark_runner/benchmark_operator/benchmark_operator_workloads.py:262]: Thx for the input, update it in last commit.

File: benchmark_runner/benchmark_operator/workload_flavors/func_ci/uperf/uperf_data_template.yaml [hunk 8]
```diff
@@ -18,7 +18,7 @@ shared_data:
   size3: 1024
   size4: 4096
   size5: 16384
-  nthr: 1
+  nthr: 2
```
[the remark this answers is not in the document]
Comment (ebattat) [reply benchmark_runner/benchmark_operator/workload_flavors/func_ci/uperf/uperf_data_template.yaml:21]: I have added num_threads 1,2

File: benchmark_runner/benchmark_operator/workload_flavors/perf_ci/hammerdb/hammerdb_data_template.yaml [hunk 9]
```diff
@@ -0,0 +1,45 @@
+shared_data:
```
[the remark this answers is not in the document]
Comment (ebattat) [reply benchmark_runner/benchmark_operator/workload_flavors/perf_ci/hammerdb/hammerdb_data_template.yaml:1]: The parameters under shared_data are relevant for pod and vm yamls.

File: benchmark_runner/benchmark_operator/benchmark_operator_workloads.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (RobertKrawitz) [reply benchmark_runner/benchmark_operator/benchmark_operator_workloads.py:23]: Right, but it's now holding more than just environment variables by the look of it.

File: .github/workflows/CI.yml [hunk 10]
```diff
@@ -100,12 +100,12 @@ jobs:
         ELASTICSEARCH: ${{ secrets.ELASTICSEARCH_CI }}
         ELASTICSEARCH_PORT: ${{ secrets.ELASTICSEARCH_PORT_CI }}
       run: |
-        BUILD_VERSION=$(python3 setup.py --version)
+        build_version=$(python3 setup.py --version)
         declare -a WORKLOADS=('stressng_pod' 'stressng_vm')
         for WORKLOAD in "${WORKLOADS[@]}"
```
Comment (RobertKrawitz) [inline .github/workflows/CI.yml:105]: Still need to fix this one.

File: .github/workflows/Nightly_Func_CI.yml [hunk 11]
```diff
@@ -32,35 +32,53 @@ jobs:
...
+                benchmark_operator_id=$(git rev-parse @)
+            else
+                benchmark_wrapper_id=$(git rev-parse @)
+            fi
+            cd -
+        done
+        build=$(pip freeze | grep benchmark-runner | sed 's/==/=/g')
+        build_version="$(cut -d'=' -f2 <<<"$build")"
+        start=$(printf '%(%s)T' -1)
         declare -a WORKLOADS=('stressng_pod' 'stressng_vm' 'uperf_pod' 'uperf_vm' 'hammerdb_pod_mariadb' 'hammerdb_vm_mariadb' 'hammerdb_pod_postgres' 'hammerdb_vm_postgres' 'hammerdb_pod_mssql' 'hammerdb_vm_mssql')
```
Comment (RobertKrawitz) [inline .github/workflows/Nightly_Func_CI.yml:57]: Variable naming (lower case)

File: benchmark_runner/benchmark_operator/workload_flavors/perf_ci/hammerdb/hammerdb_data_template.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (ebattat) [reply benchmark_runner/benchmark_operator/workload_flavors/perf_ci/hammerdb/hammerdb_data_template.yaml:1]: Sorry, did not get you. what do you mean in old one ?

File: .github/workflows/Nightly_Func_CI.yml (hunk 11 above)
Comment (ebattat) [reply .github/workflows/Nightly_Func_CI.yml]: Done

Comment (ebattat) [reply .github/workflows/Nightly_Func_CI.yml]: @RobertKrawitz, Please let me know if there are more things.

File: .github/workflows/Nightly_Perf_CI.yml [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+        cd -
+        # get last commit id of benchmark-wrapper
+        git clone https://github.com/cloud-bulldozer/benchmark-wrapper $RUNNER_PATH/benchmark-wrapper
+        cd $RUNNER_PATH/benchmark-wrapper
+        benchmark_wrapper=$(git log --format="%H" -n 1)
+        cd -
+        BUILD=$(pip freeze | grep benchmark-runner | sed 's/==/=/g')
+        BUILD_VERSION="$(cut -d'=' -f2 <<<"$BUILD")"
+        start=`date +%s`
+        declare -a WORKLOADS=('stressng_pod' 'stressng_vm' 'uperf_pod' 'uperf_vm' 'hammerdb_pod_mariadb' 'hammerdb_vm_mariadb' 'hammerdb_pod_postgres' 'hammerdb_vm_postgres' 'hammerdb_pod_mssql' 'hammerdb_vm_mssql')
```
[the remark this answers is not in the document]
Comment (ebattat) [reply .github/workflows/Nightly_Perf_CI.yml]: Fixed in last commit

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]

File: .github/workflows/Nightly_Func_CI.yml [hunk 13]
```diff
@@ -32,35 +32,53 @@ jobs:
...
+            echo '>>>>>>>>>>>>>>>>>>>>>>>>>> End E2E workload:' $workload '>>>>>>>>>>>>>>>>>>>>>>>>>>>>'
+        done
+        end=$(printf '%(%s)T' -1)
+        ci_minutes_time=$(($((end-start))/60))
+        echo '>>>>>>>>>>>>>>>>>>>>>>>>>> Update CI status pass >>>>>>>>>>>>>>>>>>>>>>>>>>'
+        podman run --rm -e WORKLOAD=$WORKLOAD -e KUBEADMIN_PASSWORD=$KUBEADMIN_PASSWORD -e PIN_NODE_BENCHMARK_OPERATOR=$PIN_NODE_BENCHMARK_OPERATOR -e PIN_NODE1=$PIN_NODE1 -e PIN_NODE2=$PIN_NODE2 -e ELASTICSEARCH=$ELASTICSEARCH -e ELASTICSEARCH_PORT=$ELASTICSEARCH_PORT -e BUILD_VERSION=$build_version -e CI_STATUS=pass -e CI_MINUTES_TIME=$ci_minutes_time -e BENCHMARK_OPERATOR_ID=$benchmark_operator_id -e BENCHMARK_WRAPPER_ID=$benchmark_wrapper_id -e log_level=INFO -v $KUBECONFIG:/root/.kube/config --privileged quay.io/ebattat/benchmark-runner:v$build_version
```
[the remark this answers is not in the document]
Comment (ebattat) [reply .github/workflows/Nightly_Func_CI.yml:67]: I prefer not change it, its  according the recommendation

File: .github/workflows/CI.yml [hunk 14]
```diff
@@ -105,7 +105,7 @@ jobs:
         for workload in "${workloads[@]}"
         do
             echo '>>>>>>>>>>>>>>>>>>>>>>>>>> Start E2E workload:' $workload '>>>>>>>>>>>>>>>>>>>>>>>>>>'
-            podman run --rm -e WORKLOAD=$workload -e KUBEADMIN_PASSWORD=$KUBEADMIN_PASSWORD -e PIN_NODE_BENCHMARK_OPERATOR=$PIN_NODE_BENCHMARK_OPERATOR -e PIN_NODE1=$PIN_NODE1 -e PIN_NODE2=$PIN_NODE2 -e ELASTICSEARCH=$ELASTICSEARCH -e ELASTICSEARCH_PORT=$ELASTICSEARCH_PORT -e BUILD_VERSION=$build_version -e log_level=INFO -v $KUBECONFIG:/root/.kube/config --privileged quay.io/ebattat/benchmark-runner:v$build_version
+            podman run --rm -e WORKLOAD="$workload" -e KUBEADMIN_PASSWORD="$KUBEADMIN_PASSWORD" -e PIN_NODE_BENCHMARK_OPERATOR="$PIN_NODE_BENCHMARK_OPERATOR" -e PIN_NODE1="$PIN_NODE1" -e PIN_NODE2="$PIN_NODE2" -e ELASTICSEARCH="$ELASTICSEARCH" -e ELASTICSEARCH_PORT="$ELASTICSEARCH_PORT" -e BUILD_VERSION="$build_version" -e log_level=INFO -v $KUBECONFIG:/root/.kube/config --privileged quay.io/ebattat/benchmark-runner:v$build_version
```
Comment (RobertKrawitz) [inline .github/workflows/CI.yml:108]: Also need to protect KUBECONFIG and build_version.  Same applies to Nightly_Func_CI.yml and Nightly_Perf_CI.yml

**Review by RobertKrawitz — CHANGES_REQUESTED** [review]

File: .github/workflows/CI.yml (hunk 14 above)
Comment (ebattat) [reply .github/workflows/CI.yml]: Fixed in last commit

**Review by RobertKrawitz — APPROVED** [review]

**Comment by RobertKrawitz** [comment]
/lgtm
/approve

Go ahead and squash the commits and merge.

**Pull request merged**